src/Microsoft.AspNetCore.Blazor.Server/AutoRebuild/AutoRebuildExtensions.cs

```diff
@@ -77,6 +109,12 @@ public static void UseAutoRebuild(this IApplicationBuilder appBuilder, BlazorCon
             });
         }
 
+        private static string FindFirstFileInDirectoryRecursive(string rootDirectory, string pattern)
```

Comment:
Is https://msdn.microsoft.com/en-us/library/dd383571(v=vs.110).aspx a suitable replacement for this or is something non-obvious happening here?

Reply:
I've checked the implementation and now ```Directory``` is backed by FileSystemEntry* stuff, which is really cool, the whole recursive file search could be a single liner with (static) options:

```
var enumerationOptions = new EnumerationOptions
{
    IgnoreInaccessible = true,
    RecurseSubdirectories = true
};

var fileToTouch = Directory.GetFiles(hostAppContentRootPath, "*.cs", enumerationOptions).FirstOrDefault();
```

Reply:
I would have thought that `RecurseSubdirectories = true` would force it to build a list containing every single `.cs` file in the entire directory hierarchy, of which there might be many. I was trying to let to stop as soon as it finds any directory containing any `.cs` file.

Reply:
Fortunately not, if you look at the implementation the same API is used with a different predicate:

https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/Directory.cs#L160

Reply:
From EnumerateFiles remarks: 

> The EnumerateFiles and GetFiles methods differ as follows: When you use EnumerateFiles, you can start enumerating the collection of names before the whole collection is returned; when you use GetFiles, you must wait for the whole array of names to be returned before you can access the array. Therefore, when you are working with many files and directories, EnumerateFiles can be more efficient.

Reply:
Yup, that's true, but the single liner above by changing ```GetFiles``` to ```EnumerateFiles``` would still remove the need for Linq and others.

Reply:
`Directory.EnumerateFiles` is a good fit here, so have switched to that. Thanks for the tip.